dali/operators/reader/loader/webdataset_loader.h

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_LOADER_WEBDATASET_LOADER_H_
+#define DALI_OPERATORS_READER_LOADER_WEBDATASET_LOADER_H_
+
+#include <fstream>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+#include <utility>
+#include <memory>
+#include "dali/operators/reader/loader/loader.h"
+#include "dali/operators/reader/loader/webdataset/tar_utils.h"
+#include "dali/pipeline/data/tensor.h"
+
+namespace dali {
+namespace detail {
+namespace wds {
+
```

Comment:
Maybe we should consider swapping this?
```c++
namespace wds{
namespace detail{

}
class WebdatasetReader;
} // namespace wds

Reply:
I don't know, all of the implementation details are usually in the detail namespace, and I guess it wouldn't make much sense to create a whole separate namespace just for a single class. How do others feel about that?

Reply:
I agree with the current order. wds seems to be just a details. Other way around wds would have only details inside what doesn't make much sense IMHO.

Reply:
Or maybe you should get rid of wds namespace at all? What value it adds now?

Reply:
prevents duplicates in detail namespace (there is quite a lot of stuff from wds in detail)

Reply:
In defense of current order:
1. When inside `wds::detail`, it's hard to reach to other `detail` namespaces.
2. Separating `wds` is nice, because `detail` namespaces tend to contain utility functions with easily-colliding names.
Having said that, it could just be one level of namespace, like `wds_impl`.